src/librustdoc/html/render/span_map.rs

```diff
@@ -189,6 +189,27 @@ impl SpanMapVisitor<'_> {
             self.matches.insert(span, link);
         }
     }
+
+    fn handle_pat(&mut self, p: &Pat<'_>) {
```

Comment:
This function very closely mirrors `intravisit::walk_pat`. The only difference is the QPath handling. The more general and cleaner solution would be to explicitly implement `visit_qself`. This would automatically handle paths in expression, pattern and type contexts.

For example, rn we don't linkify `Self::Var` (`ExprKind::Path(qpath)`):

````rs
impl Trait for Ty { pub fn f() { let _ = Self::Var; } }
pub enum Ty { Var }
pub trait Trait { fn f(); }
````

Moreover, I presume we don't linkify `Self::Assoc` in `impl ... for ... { type Ty = Self::Assoc; }` (`TyKind::Path(qpath)`) but I can't check, I'm commuting atm.

However, this would require this "`maybe_typeck_results`/`qpath_res` " setup I mentioned on Zulip. It would also understand fully-qualified paths `<Ty as TraitRef>::assoc`.

Anyway, if you don't want to do that in this PR, I can send a follup-up PR that impls that.

My 2¢ – won't say more rn, I'm still commuting lol.

Reply:
It was not in hurry for review you know. :sweat_smile: (But thanks!)

So do you want me to implement `visit_qself` in this PR or in a follow-up?

Reply:
Let me get back home, then I'm going to answer that xD